### PR TITLE
RL renaming

### PR DESCRIPTION
**Pull request by lihuoran**
# Description

- Rename `Trainer` to `AbsAlgorithm`
- Rename `TrainerManager` to `TrainingManager`
- Change all "Trainer" in the doc to "Algorithm instance". Change all "trainer" in the code to "algo_inst".
- Update related docs/notebooks.
- Refine some RL related interface names (for example, `step()` => `train_step()`)

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: docs/source/key_components/rl_toolkit.rst [hunk 1]
```diff
@@ -129,29 +129,29 @@ There are also some other models for training purposes. For example:
...
-Trainer
+Algorithm
 -------
 
 When introducing policies, we mentioned that policies cannot train themselves. Instead, they have to be trained
-by external trainers. In MARO, a trainer represents an RL algorithm, such as DQN, actor-critic,
-and so on. Trainers take interaction experiences and store them in the internal memory, and then use the experiences
-in the memory to train the policies. Like ``RLPolicy``, trainers are also concrete classes, which means they could
-be used by configuring parameters. Currently, we have 4 trainers (algorithms) in MARO:
+by external algorithms.
```
Comment (Jinyu-W) [inline docs/source/key_components/rl_toolkit.rst:138]: Still strange by saying `train by algorithm`...

**Review by Jinyu-W — COMMENTED** [review]
How about `Algorithm` -> `Algorithm Trainer`? @lihuoran  @ysqyang

File: maro/rl/rollout/env_sampler.py [hunk 2]
```diff
@@ -155,15 +155,15 @@ def agent_names(self) -> list:
     def num_agents(self) -> int:
         return len(self.agent_state_dict)
 
-    def split_contents(self, agent2trainer: Dict[Any, str]) -> Dict[str, ExpElement]:
-        """Split the ExpElement's contents by trainer.
+    def split_contents(self, agent_to_algo_inst: Dict[Any, str]) -> Dict[str, ExpElement]:
```
Comment (Jinyu-W) [inline maro/rl/rollout/env_sampler.py:158]: Is there any other mappings with name not following `aaa2bbb` ?
If not, I think maybe it is better to align the naming pattern, e.g. `agent2algoInst`/`agent2algoTrainer` ?

File: maro/rl/training/__init__.py [hunk 3]
```diff
@@ -1,18 +1,18 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+from .abs_algorithm import AbsAlgorithm, AlgorithmParams, MultiAlgorithm, SingleAlgorithm
```
Comment (Jinyu-W) [inline maro/rl/training/__init__.py:4]: `MultiAlgorithm` a little bit strange = =
Actually it is a trainer for multi-agent algorithm?

File: maro/rl/rollout/env_sampler.py (hunk 2 above)
Comment (lihuoran) [reply maro/rl/rollout/env_sampler.py:158]: It is not recommended to use upper class letters in the Python variables. Just change it to `agent2algoinst` anyway.

File: maro/rl/training/__init__.py (hunk 3 above)
Comment (lihuoran) [reply maro/rl/training/__init__.py:4]: `SingleAlgorithm` => `SingleAgentAlgorithm`
`MultiAlgorithm` => `MultiAgentAlgorithm`

**Pull request closed without merging**